tests/Functional/Driver/PDO/MySQL/DeadlockTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Doctrine\DBAL\Tests\Functional\Driver\PDO\MySQL;
+
+use Doctrine\DBAL\Exception\DeadlockException;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Tests\TestUtil;
+
+use function implode;
+use function pcntl_fork;
+use function sleep;
+use function sprintf;
+
+class DeadlockTest extends FunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $supportedDrivers = ['pdo_mysql', 'mysqli', 'mssql'];
```

Comment:
The doesn't seem to be a `mssql` driver, and the corresponding exception converter doesn't throw a deadlock exception. At the same time, the test seems to pass with `pdo_pgsql`.